Automattic/wp-calypso
# A4A: Refactor expertise form.

This PR modifies the Partner directory expertise form to use the official data structure and the existing 'Services' and 'Products' selector component.

## Proposed Changes

* Update useExpertiseForm to use the `AgencyDirectoryApplication` type as the formData type. Update some functions to match the new data structure and remove unnecessary constants to represent fields.
* Update the Expertise form component to use the **'Services'** and **'Products'** selector component.
* Add initial CTA functionality in the Expertise form.
* Make the Expertise form aware of initial data.

## Testing Instructions


* Use the A4A live link and go to `/partner-directory/agency-expertise`
* Confirm that the form is working as expected.
   * Confirm the Services and Products field uses the new selector.
   * Confirm that the CTA button (disable state) works as expected.


## Pre-merge Checklist
